pypeclub/OpenPype
# Ftrack: Private project server actions

## Issue
Ftrack event server may not have access to private projects which may be confusing for project manager and it may crash few event handlers.

## Changes
- server actions should not crash if entities are not queried from private project
- added action `Missing permissions` which is shown only if someone tries to show actions on private project for which server api key does not have permissions

## Notes
If you don't like messages or titles in new action don't ask and modify :)

## How to test
Easiest is with using api API key.
- create new api API key in ftrack (with API role)
![image](https://user-images.githubusercontent.com/43494761/128213248-698a39f6-9a15-49d8-a1f6-dae5a667c7c9.png)
- run ftrack event server using the api key
- create private project and try to show actions on it
- you should see action `Missing permissions`
![image](https://user-images.githubusercontent.com/43494761/128213500-27ac72f3-7eba-42b5-8720-a2d0d3eae75c.png)
